provider/cloudflare/cloudflare.go

```diff
@@ -394,34 +385,37 @@ func (p *CloudFlareProvider) submitChanges(ctx context.Context, changes []*cloud
 			}
 
 			resourceContainer := cloudflare.ZoneIdentifier(zoneID)
+			records, err := p.listDNSRecordsWithAutoPagination(ctx, zoneID)
+			if err != nil {
+				return fmt.Errorf("could not fetch records from zone, %v", err)
```

Comment:
not covered with unit test 

![Screenshot 2025-03-06 at 07 46 10](https://github.com/user-attachments/assets/873ff5fa-d27e-4736-882b-a1730ddeba44)

Reply:
covered